moirae/estimators/online/dual.py

```diff
@@ -0,0 +1,172 @@
+""" Framework for dual estimation of transient vector and A-SOH"""
+from typing import Union, Tuple, Optional, TypedDict, Literal
+from typing_extensions import Self
+
+import numpy as np
+
+from moirae.models.base import InputQuantities, OutputQuantities, GeneralContainer, HealthVariable, CellModel
+from .utils.model import CellModelWrapper, DegradationModelWrapper, convert_vals_model_to_filter, convert_numpy_to_model
+from moirae.estimators.online import OnlineEstimator
+from .filters.base import BaseFilter
+from .filters.distributions import MultivariateRandomDistribution, MultivariateGaussian
+from .filters.kalman.unscented import UnscentedKalmanFilter as UKF
+
+
+class individual_UKF_tuning_inputs(TypedDict):
```

Comment:
Style point: Classes should be named with PascalCase not snake_case. Could you change this and the next class to the capitalized style?

We should also move the "UKFTuningInputs" to the same module as the UKF to keep things grouped together better (e.g., if you want to use the class for the joint or other frameworks).

Reply:
We should consider having all our future filters implementations use a similar style of coming packed with their own TypedDict. Kind of nice on the user to clearly see what kind of kwargs they'll need to pass to other constructors.

Reply:
Good idea! Moved this class to the filter location and renamed it.